google_photos_takeout_helper/__main__.py

```diff
@@ -174,7 +176,44 @@ def get_hash(file: Path, first_chunk_only=False, hash_algo=_hashlib.sha1):
                     hashobj.update(chunk)
         return hashobj.digest()
 
-    # PART 1: removing duplicates
+    def populate_album_map(path: Path, filter_fun=lambda f: (is_photo(f) or is_video(f))):
+        if not path.is_dir():
+            raise NotADirectoryError('populate_album_map only handles directories not files')
+        try:
+            meta_file_exists = find_album_meta_json_file(path)
+
+            if meta_file_exists:  # means that we are processing an album so process
+                for file in path.rglob("*"):
+                    if file.is_file() and filter_fun(file):
+                        file_name = file.name
+                        if not Path(str(FIXED_DIR) + "/" + file.name).is_file():
+                            try:
+                                full_hash = get_hash(file, first_chunk_only=False)
+                                if full_hash in files_by_full_hash:
+                                    full_hash_files = files_by_full_hash[full_hash]
+                                    if len(full_hash_files) != 1:
+                                        print(
+                                            "full_hash_files list should only be one after duplication removal, bad state")
+                                        exit()
```

Comment:
You've inserted `exit()` here, but because there is no code number, it doesn't (i think so)

I'm already heavliy editing this part, will want you to review after I'm done